### PR TITLE
refactor: rename errors

### PR DESCRIPTION
**Pull request by lzap**
It is a good practice to prefix error values with `Err` and suffix error types:

```
// error type
type MyErr error

// error value
var ErrMy = errors.New("error")
```

This PR is pure refactoring, no changes introduced.

### CODE REVIEW

**Comment by avitova** [comment]
Good job in finding all of these! :+1: 
I've found one more place. Could you please change this as well?
[account.go](https://github.com/RHEnVision/provisioning-backend/blob/168476c45aca794e59ea33eef851d8471b306445/internal/identity/account.go#L14)

File: internal/clients/http/errors.go [hunk 1]
```diff
@@ -8,32 +8,32 @@ import (
...
-	NotEvenErr                          = errors.New("number of keys and values is not even when building a query")
-	SourcesInvalidAuthentication        = usrerr.New(400, "insufficient data for authentication", "")
+	ErrApplicationNotFound              = usrerr.New(404, "application not found is sources", "")
+	ErrApplicationTypeNotFound          = usrerr.New(404, "application type 'provisioning' not found in sources", "")
+	ErrSourceNotFound                   = usrerr.New(404, "source not found", "")
+	ErrAuthenticationSourceAssociation  = usrerr.New(404, "authentication associated to source id not found", "")
+	ErrAuthenticationForSourcesNotFound = usrerr.New(404, "authentications for source weren't found in sources", "")
+	ErrApplicationRead                  = usrerr.New(500, "application read returned no application type in sources", "")
+	ErrSourceTypeNameNotFound           = usrerr.New(404, "source type name not found", "")
+	ErrNotEven                          = errors.New("number of keys and values is not even when building a query")
```
Comment (avitova) [inline internal/clients/http/errors.go:18]: Hmm, why does this not use the usrerr package? :thinking:

File: internal/clients/http/errors.go [hunk 2]
```diff
@@ -8,32 +8,32 @@ import (
...
-	DuplicatePubkeyErr             = usrerr.New(406, "public key already exists in target cloud provider account and region", "")
-	PubkeyNotFoundErr              = usrerr.New(404, "pubkey not found in AWS account", "")
-	ServiceAccountUnsupportedOpErr = usrerr.New(500, "unsupported operation on service account", "")
-	ARNParsingError                = usrerr.New(500, "ARN parsing error", "")
-	NoReservationErr               = usrerr.New(404, "no reservation has found in AWS response", "")
+	ErrDuplicatePubkey             = usrerr.New(406, "public key already exists in target cloud provider account and region", "")
+	ErrPubkeyNotFound              = usrerr.New(404, "pubkey not found in AWS account", "")
+	ErrServiceAccountUnsupportedOp = usrerr.New(500, "unsupported operation on service account", "")
+	ErrARNParsing                  = usrerr.New(500, "ARN parsing error", "")
+	ErrNoReservation               = usrerr.New(404, "no reservation has found in AWS response", "")
```
Comment (avitova) [inline internal/clients/http/errors.go:38]:
```suggestion
	ErrNoReservation               = usrerr.New(404, "no reservation was found in AWS response", "")
```

File: internal/kafka/kafka.go [hunk 3]
```diff
@@ -33,8 +33,8 @@ type kafkaBroker struct {
 var _ Broker = &kafkaBroker{}
 
 var (
-	DifferentTopicErr       = errors.New("messages in batch have different topics")
-	UnknownSaslMechanismErr = errors.New("unknown SASL mechanism")
+	ErrDifferentTopic       = errors.New("messages in batch have different topics")
+	ErrUnknownSaslMechanism = errors.New("unknown SASL mechanism")
```
Comment (avitova) [inline internal/kafka/kafka.go:37]:
```suggestion
	ErrUnknownSASLMechanism = errors.New("unknown SASL mechanism")
```

File: internal/kafka/kafka.go [hunk 4]
```diff
@@ -59,7 +59,7 @@ func createSASLMechanism(saslMechanismName string, username string, password str
 
 		return mechanism, nil
 	default:
-		return nil, fmt.Errorf("%w: %s", UnknownSaslMechanismErr, saslMechanismName)
+		return nil, fmt.Errorf("%w: %s", ErrUnknownSaslMechanism, saslMechanismName)
```
Comment (avitova) [inline internal/kafka/kafka.go:62]:
```suggestion
		return nil, fmt.Errorf("%w: %s", ErrUnknownSASLMechanism, saslMechanismName)
```

**Review by avitova — COMMENTED** [review]
There are a few nitpicks, but I do not insist. TY

**Comment by lzap** [comment]
Fixed everything. Converted one error to panic. Removed unused errors. Three commits.

**Comment by ezr-ondrej** [comment]
It seems we have a test for the BuildQuery error :) we'll need to fix the test now

File: internal/clients/http/errors.go (hunk 1 above)
Comment (avitova) [reply internal/clients/http/errors.go]: Cool, thanks for the explanation. :)

**Comment by avitova** [comment]
Do you want to squash some of the commits before merge? @lzap

**Comment by lzap** [comment]
I think keeping it like that in the history can be useful, but if you prefer feel free to squash on merge.

**Comment by lzap** [comment]
Actually, I am adding one more commit. The reason is that I am looking at GlitchTip report `:get source authentication call: backend service returned unexpected HTTP code` and I would be interested if the result code was 404 (no authorization) or some different result. The new commit adds HTTP code into the message as well as unifies wording across other source messages.

**Comment by lzap** [comment]
And one more commit which fixes an ugly fingerprint being present in the error message (glitchtip cannot group errors together):

image

File: internal/jobs/launch_instance_aws.go [hunk 5]
```diff
@@ -147,14 +147,15 @@ func DoEnsurePubkeyOnAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
...
+		if errors.Is(err, http.ErrPubkeyNotFound) {
 			pkr.Tag = ""
 			pkr.RandomizeTag()
 			pkr.Handle, err = ec2Client.ImportPubkey(ctx, pubkey, pkr.FormattedTag())
 
-			if errors.Is(err, http.DuplicatePubkeyErr) {
+			if errors.Is(err, http.ErrDuplicatePubkey) {
 				// key not found by fingerprint but importing failed for duplicate err so fingerprints do not match
-				return fmt.Errorf("key with fingerprint %s not found on AWS, but importing the key failed: %w", pubkey.Fingerprint, err)
+				logger.Warn().Msgf("Pubkey with fingerprint %s not found on AWS, but importing the key failed: %w", pubkey.Fingerprint, err)
```
Comment (ezr-ondrej) [inline internal/jobs/launch_instance_aws.go:157]: Linter is right, `%w` wouldn't play well with `Msgf` I guess :)

**Review by ezr-ondrej — COMMENTED** [review]
There's one linter issue in the last commit, but apart of it, looks great to me, I'd love to get it in tomorrow/friday so we can soak it a bit in stage until the last push :)

**Review by ezr-ondrej — COMMENTED** [review]
There's one linter issue in the last commit, but apart of it, looks great to me, I'd love to get it in tomorrow/friday so we can soak it a bit in stage until the last push :)

**Comment by lzap** [comment]
Sorry for the delay, resolved.

**Comment by lzap** [comment]
We need to refresh images I guess?

**Comment by ezr-ondrej** [comment]
> We need to refresh images I guess?

Yup and should be done if I got it correctly :) lets see

**Comment by ezr-ondrej** [comment]
/retest
deployment is unhappy again :(

**Review by avitova — APPROVED** [review]
The tests are happy, and so am I!:) TY, @ezr-ondrej care to merge?

**Comment by lzap** [comment]
🙏🏻 Thanks.

**Review by ezr-ondrej — APPROVED** [review]

**Pull request merged**